service/gcs/main.go

```diff
@@ -18,8 +20,24 @@ import (
 	oci "github.com/opencontainers/runtime-spec/specs-go"
 	"github.com/sirupsen/logrus"
 	"go.opencensus.io/trace"
+	"golang.org/x/sys/unix"
 )
 
+func readMemoryEvents(efd uintptr, cgName string, limit int64) {
+	// Buffer must be >= 8 bytes for eventfd reads
+	// http://man7.org/linux/man-pages/man2/eventfd.2.html
+	buf := make([]byte, 8)
+	for {
+		_, err := cgrouputils.ReadEventFD(efd, buf)
+		if err != nil {
+			logrus.WithError(err).Warn("opengcs::readEvents - failed to read from eventfd")
+			unix.Close(int(efd))
```

Comment:
Should this just be a defer at the top of the function?

It's a little weird that this function takes ownership of the fd; that should be made clear in a comment.

Reply:
I'll just defer the file closures in main instead, I agree it's odd